pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -847,6 +878,188 @@ func validatePort(node *model.Proxy, i int, bindToPort bool) bool {
 	return proxyProcessUID == "0"
 }
 
+func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPListenerForPortOrUDS(listenerMapKey *string,
+	currentListenerEntry **outboundListenerEntry, listenerOpts *buildListenerOpts,
+	pluginParams *plugin.InputParams, listenerMap map[string]*outboundListenerEntry, actualWildcard string) (bool, []*filterChainOpts) {
+	// first identify the bind if its not set. Then construct the key
+	// used to lookup the listener in the conflict map.
+	if len(listenerOpts.bind) == 0 { // no user specified bind. Use 0.0.0.0:Port
+		listenerOpts.bind = actualWildcard
+	}
+	*listenerMapKey = fmt.Sprintf("%s:%d", listenerOpts.bind, pluginParams.Port.Port)
+
+	var exists bool
+
+	// Have we already generated a listener for this Port based on user
+	// specified listener ports? if so, we should not add any more HTTP
+	// services to the port. The user could have specified a sidecar
+	// resource with one or more explicit ports and then added a catch
+	// all listener, implying add all other ports as usual. When we are
+	// iterating through the services for a catchAll egress listener,
+	// the caller would have set the locked bit for each listener Entry
+	// in the map.
+	//
+	// Check if this HTTP listener conflicts with an existing TCP
+	// listener. We could have listener conflicts occur on unix domain
+	// sockets, or on IP binds. Specifically, its common to see
+	// conflicts on binds for wildcard address when a service has NONE
+	// resolution type, since we collapse all HTTP listeners into a
+	// single 0.0.0.0:port listener and use vhosts to distinguish
+	// individual http services in that port
+	if *currentListenerEntry, exists = listenerMap[*listenerMapKey]; exists {
+		// NOTE: This is not a conflict. This is simply filtering the
+		// services for a given listener explicitly.
+		if (*currentListenerEntry).locked {
+			return false, nil
+		}
+		if pluginParams.Service != nil {
+			if !(*currentListenerEntry).servicePort.Protocol.IsHTTP() {
+				outboundListenerConflict{
+					metric:          model.ProxyStatusConflictOutboundListenerTCPOverHTTP,
+					node:            pluginParams.Node,
+					listenerName:    *listenerMapKey,
+					currentServices: (*currentListenerEntry).services,
+					currentProtocol: (*currentListenerEntry).servicePort.Protocol,
+					newHostname:     pluginParams.Service.Hostname,
+					newProtocol:     pluginParams.Port.Protocol,
+				}.addMetric(pluginParams.Push)
```

Comment:
Don’t you need to fix this logic by adding the detection? That’s the whole point of this sniffing

Reply:
I set the port protocol of last egress listener as `auto` instead of using service port protocol.